tsdb/head_test.go

```diff
@@ -5219,3 +5351,45 @@ func TestSnapshotAheadOfWALError(t *testing.T) {
 
 	require.NoError(t, head.Close())
 }
+
+type sampleTypeScenario struct {
+	sampleType string
+	appendFunc func(appender storage.Appender, lbls labels.Labels, ts int64, value int64) (storage.SeriesRef, error, sample)
+	sampleFunc func(ts, value int64) sample
+}
+
+var sampleTypeScenarios = map[string]sampleTypeScenario{
```

Comment:
I like this refactoring so we can re-use the same appendFunc and sampleFunc for the different data types